packages/instantsearch.js/src/types/widget.ts

```diff
@@ -134,6 +137,30 @@ export type WidgetDescription = {
   indexUiState?: Record<string, unknown>;
 };
 
+type SearchWidgetLifeCycle<TWidgetDescription extends WidgetDescription> = {
+  dependsOn?: 'search';
+  getWidgetParameters?: (
+    state: SearchParameters,
+    widgetParametersOptions: {
+      uiState: Expand<
+        Partial<TWidgetDescription['indexUiState'] & IndexUiState>
+      >;
+    }
```

Comment:
No biggie but this is common with `RecommendWidgetLifeCycle`. We could make a generic type so new APIs are even easier to support for new contributors.

Reply:
Not sure they would really be the same, I added `uiState` but I don't think it would ever be used for recommend. Still too early to abstract further imo but if there ever is a 3rd type or we need more params then yes

Reply:
Also I left it as optional for `search` when it's required for `recommend` so we can't really make it generic for now